### PR TITLE
BTM 735 Changed status colours

### PR DESCRIPTION
**Pull request by Xfarya**
Implements the status colour updates described in BTM-735.

(MarkS says:) The significant refactor here was to rein in all the separate lookup methods we used to have for these fields -- now, there are just enums for 'InvoiceStatuses' and 'LineItemStatuses' (closely related), and all the information needed to display them is on their associated objects.

### CODE REVIEW

File: src/frontend/extract-helpers/get-invoice-status.test.ts [hunk 1]
```diff
@@ -114,23 +126,25 @@ describe("getInvoiceBanner", () => {
...
     // Act
-    const result = getInvoiceBanner(givenLineItems);
+    const result = getInvoiceStatus(givenLineItems);
     // Assert
     expect(result).toEqual({
-      bannerClass: "warning",
-      status: "Unable to find rate",
+      bannerClass: "error",
+      bannerText: "Unable to find rate",
+      statusLabel: statusLabelLookup[StatusLabels.error],
```
[the remark this answers is not in the document]
Comment (markschnitziusgovuk) [reply src/frontend/extract-helpers/get-invoice-status.test.ts:147]: Not sure where you are seeing that -- the ticket shows this to be red ERROR and hasn't been changed recently?

File: src/frontend/extract-helpers/get-invoice-status.test.ts [hunk 2]
```diff
@@ -166,11 +182,12 @@ describe("getInvoiceBanner", () => {
...
     // Act
-    const result = getInvoiceBanner(givenLineItems);
+    const result = getInvoiceStatus(givenLineItems);
     // Assert
     expect(result).toEqual({
-      bannerClass: "error",
-      status: "Invoice above threshold",
+      bannerClass: "warning",
+      bannerText: "Invoice above threshold",
+      statusLabel: statusLabelLookup[StatusLabels.aboveThreshold],
```
[the remark this answers is not in the document]
Comment (markschnitziusgovuk) [reply src/frontend/extract-helpers/get-invoice-status.test.ts]: Fixed now.

File: src/frontend/extract-helpers/get-reconciliation-rows.test.ts [hunk 3]
```diff
@@ -161,10 +157,7 @@ describe("getReconciliationRows", () => {
         quantityDiscrepancy: "9",
         priceDiscrepancy: "£27.50",
         percentageDiscrepancy: "Unexpected invoice charge",
-        status: {
-          message: "Error",
-          class: "govuk-tag--grey",
-        },
+        status: lineItemStatusLookup[LineItemStatuses.unexpectedCharge],
```
[the remark this answers is not in the document]
Comment (markschnitziusgovuk) [reply src/frontend/extract-helpers/get-reconciliation-rows.test.ts:160]: Ticket says grey to yellow for this -- not sure where you are looking.

[the remark this answers is not in the document]
Comment (markschnitziusgovuk) [reply src/frontend/extract-helpers/get-reconciliation-rows.test.ts:160]: Yes, but 'unexpected invoice charge' is no longer an error, it's a warning.  I guess it's moot now though as the latest refactor causes the status colour and text to be looked up based on the status enum.

File: src/frontend/utils/invoice-statuses.ts [hunk 4]
```diff
@@ -15,66 +15,107 @@ export type InvoiceStatus = {
...
 }
 
+export const invoiceStatusLabelLookup: Record<InvoiceStatuses, StatusLabel> = {
+  [InvoiceStatuses.invoiceAndEventsMissing]: {
+    message: "Pending",
+    class: "govuk-tag--blue",
+  },
+  [InvoiceStatuses.invoiceDataMissing]: {
+    message: "Warning",
+    class: "govuk-tag--blue",
```
[the remark this answers is not in the document]
Comment (Xfarya) [reply src/frontend/utils/invoice-statuses.ts:35]: changed to "Pending"

File: src/frontend/extract-helpers/get-invoice-status.test.ts [hunk 5]
```diff
@@ -237,8 +242,8 @@ describe("getInvoiceBanner", () => {
     // Assert
     expect(result).toEqual({
       bannerClass: "payable",
-      bannerText: "Invoice has no charge",
-      statusLabel: statusLabelLookup[StatusLabels.noCharge],
+      bannerText: "No charge",
```
[the remark this answers is not in the document]
Comment (Xfarya) [reply src/frontend/extract-helpers/get-invoice-status.test.ts:245]: reverted, thanks!

File: src/frontend/utils/line-item-statuses.ts [hunk 6]
```diff
@@ -26,43 +26,79 @@ export enum LineItemStatuses {
...
+      message: "Invoice Missing",
+      class: "govuk-tag--yellow",
+    },
+    [LineItemStatuses.eventsMissing]: {
+      message: "Events Missing",
+      class: "govuk-tag--red",
+    },
+    [LineItemStatuses.unexpectedCharge]: {
+      message: "Unexpected Charge",
+      class: "govuk-tag--yellow",
```
[the remark this answers is not in the document]
Comment (Xfarya) [reply src/frontend/utils/line-item-statuses.ts:49]: The text was wrong in the ticket and James has fixed this now - we have retained it as is

File: gds-cli
[the remark this answers is not in the document]
Comment (Xfarya) [reply gds-cli:1]: accidentally got into the pr

**Pull request merged**